includes/class-wc-payments.php

```diff
@@ -960,8 +960,10 @@ public static function platform_checkout_fields_before_billing_details() {
 	 */
 	public static function filter_woocommerce_form_field_platform_checkout_email( $field, $key, $args, $value ) {
 		$class = $args['class'][0];
-		if ( false === strpos( $class, 'platform-checkout-billing-email' ) ) {
-			$field = '';
+		if ( is_checkout() ) {
+			if ( false === strpos( $class, 'platform-checkout-billing-email' ) ) {
```

Comment:
Nit: I personally would prefer these two conditions in the same `if` block, just because it's equivalent with one less instruction:

```php
if (
    false === strpos( $class, 'platform-checkout-billing-email' ) &&
    is_checkout()
) {
```

Reply:
Good call! I can't think of any scenarios within core that would load the checkout form on the Pay for Order page, but it must be possible since [is_checkout_pay_page()](https://github.com/woocommerce/woocommerce/blob/0a381f422b94e6c9866b73d5790e2bace8186ca6/plugins/woocommerce/includes/wc-conditional-functions.php#L123) exists. I'll combine those in the same block as mentioned above.

Reply:
Just to clarify, we want to hide the old billing email field from the shortcode checkout, but not from the "pay for order" page, if it's ever possible to have a billing form there, right?

I've tested this:

- `is_checkout()` ✅ for the classic checkout page, ✅ for the pay for order page.
- `is_checkout_pay_page()` ❌ for the classic checkout page, ✅ for the pay for order page.

In that case, I guess we would want to make sure we're on the checkout page, but not on the pay for order page:

```php
if ( false === strpos( $class, 'platform-checkout-billing-email' ) && is_checkout() && ! is_checkout_pay_page() ) {
```

Reply:
>we want to hide the old billing email field from the shortcode checkout, but not from the "pay for order" page

Yep. This was a silly mistake on my part. Thanks for catching it!